src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFChart.java

```diff
@@ -527,26 +566,14 @@ public String formatRange(CellRangeAddress range) {
      * @since POI 4.0.0
      */
     private XSSFSheet getSheet() {
-        XSSFSheet sheet = null;
-        try {
-            sheet = getWorkbook().getSheetAt(0);
-        } catch (InvalidFormatException ife) {
-        } catch (IOException ioe) {
+        if(sheet==null)
+        {
+            workbook = new XSSFWorkbook();
+            sheet = workbook.createSheet();
```

Comment:
This seems to remove the ability to modify an existing chart.

Reply:
we are copying data from old sheet to this new sheet (in case of modifying chart data) in fillSheet() method, so i think this will not break any existing functionality.
we need to create new sheet because when we are writing embedded xlsx part outputstrem which we read,when writing it  in document it always writing old data into embedded excel.